### PR TITLE
CI improvements :earth_africa: 

### PR DESCRIPTION
**Pull request by oribon**
Some refactors that should save us CI resources and time:
* Creating dependencies between E2E and "basic" jobs
* Stop running everything again on merges

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -52,6 +56,22 @@ jobs:
...
           inv checkchanges --action="run inv generateapidocs"
 
+      - name: Build and save MetalLB images
+        run: |
+          inv build --binaries all
+          docker save quay.io/metallb/speaker:dev-amd64 > speaker.tar
+          docker save quay.io/metallb/controller:dev-amd64 > controller.tar
+          docker save quay.io/metallb/mirror-server:dev-amd64 > mirror-server.tar
+
+      - name: Upload images
```
Comment (fedepaol) [inline .github/workflows/ci.yaml:66]: Neat! Can we delete the artifacts as final step? I don't think we are gonna need those images for triaging failures.

**Comment by fedepaol** [comment]
I like this! Just one comment about not leaving the images in the artifacts.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (oribon) [reply .github/workflows/ci.yaml:66]: There isn't an official workflow for cleanup :confused: 
set the retention-days to 1 instead: https://github.com/actions/upload-artifact#retention-period

Comment (fedepaol) [reply .github/workflows/ci.yaml:66]: Makes sense, thanks

**Pull request merged**